Refactor the error a bit

Veetaha opened the pull request:
For the main diff see this commit: a1d41b7
The second one (`cargo fmt`) seems to have changed the line endings, so the diff looks a bit too big at the first glance.

- Use the naming according to [Rust API Guidelines](https://rust-lang.github.io/api-guidelines/naming.html)
- Drop dependency on `thiserror` - the less dependency the better!

topecongiro commented on src/lib.rs:
```diff
@@ -113,7 +115,7 @@ impl AdvisoryFileLock {
     fn drop(&mut self) {
         if let Err(err) = self.unlock() {
-            log::error!("Unlock_file failed during dropping: {}", err);
+            panic!("unlock_file failed during dropping: {}", err);
```
I prefer not to panic here, would you mind changing this back to `log::error`?

Veetaha replied:
Hmm, yeah, I don't know why I changed that

topecongiro reviewed (changes requested):
Thank you for the PR! The change LGTM overall, though I prefer not to have `panic!` in this library if possible. Do you have a reason to prefer `panic!` over `log::error`?

Veetaha commented:
@topecongiro I've rebased and removed switching to panic (df4406d), also, amended error handling for windows (2da8331)
and fixed the tests for windows (da3af7e, I guess current windows implementation is broken!?)

topecongiro reviewed (approved):
> I guess current windows implementation is broken!?

Good catch; thank you for the fix!

FYI [this blog post](https://devblogs.microsoft.com/oldnewthing/20140905-00/?p=63) was an excellent overview of the topic. I'll revisit the error handling when I add support for async files.
